lib/Driver/Compilation.cpp

```diff
@@ -291,8 +291,14 @@ namespace driver {
              "not implemented for compilations with multiple jobs");
       if (Comp.ShowJobLifecycle)
         llvm::outs() << "Added to TaskQueue: " << LogJob(Cmd) << "\n";
-      TQ->addTask(Cmd->getExecutable(), Cmd->getArguments(), llvm::None,
-                  (void *)Cmd);
+      if (Cmd->hasResponseFile()) {
+        Cmd->writeArgsToResponseFile();
+        TQ->addTask(Cmd->getExecutable(), Cmd->getResponseFileArg(),
+                    llvm::None, (void *)Cmd);
+      } else {
+        TQ->addTask(Cmd->getExecutable(), Cmd->getArguments(), llvm::None,
+                    (void *)Cmd);
+      }
```

Comment:
In fact, as I read further down, I noticed I had to stare at the two `addTask` call lines to see the difference.
Therefore I propose factoring into a small, static function that gets the arguments, or else pushing this logic into the `Cmd->getArguments` function. There should be only *one* line that calls TQ->addTask.

Reply:
`Cmd->getArguments` may be used elsewhere with the expectation that it returns the vector of arguments executed by this job, regardless of whether or not a response file is used. I felt it would be safer to have a separate method specifically for returning the response file.